tests/test-recipes/metadata/noarch_foo/meta.yaml

```diff
@@ -0,0 +1,6 @@
+package:
+  name: noarch_foo_test_package
+  version: "1.0"
+
+build:
+  noarch: foo
```

Comment:
> I think index.json is the right file, but it might be info.json.

I think index.json is the safest file.  I don't yet perfectly clear visibility on all the ways we might use this, and it's possible that the solver might need to have knowledge of noarch type at some point.  I think putting it in index.json will be a hedge so we don't limit ourselves going forward.